pkg/api/v1/service/util.go

```diff
@@ -70,7 +70,8 @@ func GetLoadBalancerSourceRanges(service *v1.Service) (utilnet.IPNetSet, error)
 // ExternalPolicyLocal checks if service has ETP = Local.
 func ExternalPolicyLocal(service *v1.Service) bool {
```

Comment:
The doc comment is incorrect; the function checks both (a) if the service has ETP=Local, _and_ (b) if the service is _allowed_ to have ETP=local. This is kind of weird... (This is a pre-existing problem with the function.)

The registry/storage code should be validating the "allowed to have ETP=local" part itself anyway, so I think this function should _just_ check `service.Spec.ExternalTrafficPolicy` (though in that case it's not clear that we even need this function).

Alternatively/additionally, we could add an `ExternallyReachable` function that checks if it is NodePort or LB or has ExternalIPs, and then use that in a bunch of places.